tools/pylib/boututils/run_wrapper.py

```diff
@@ -292,13 +292,16 @@ def launch_safe(command, *args, **kwargs):
 def build_and_log(test):
     """Run make and redirect the output to a log file. Prints input
 
-    On Windows, does nothing because executable should have already
-    been built
+    When using CMake, or on Windows, does nothing because executable
+    should have already been built
 
     """
 
     if os.name == "nt":
         return
 
+    if os.path.exists("CMakeFiles"):
+        return
```

Comment:
Do we definitely want to skip this for CMake? At the moment, at least with my setup, if a test is already built, and I go into the test directory and run `make` again, it runs, but very quickly if there is nothing to rebuild. If this is what should generally happen with CMake builds, it would be useful to keep the 'make' in `runtest` scripts, because then if you just build the library, and go into a single test directory and do `./runtest` then it just works without needing to i) build all the tests every time we build BOUT++ or ii) have to run `make; ./runtest` if you want to run a single test. It's not really important, but keeping the 'make' would remove one way that `runtest` could fail.

Reply:
The issue is really for ninja where it doesn't create a makefile or equivalent in the test directory. I originally considered `if not os.path.exists("makefile")` instead of this, but thought that might be too fragile.

It should be possible to make this work for cmake, it's just more logic here, and therefore more surface area for bugs, which is why I went with the dumb option here.

Given that this is more controversial than I first thought, I'll try and make it work!